### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by RemDelaporteMathurin**
This PR provides refactoring for the temperature and particle fluxes functions.

I did not provide tests at this stage and will keep it for another PR as I suspect a new class will be created for this

### CODE REVIEW

File: src/hisp/festim_models/mb_model.py [hunk 1]
```diff
@@ -749,3 +753,130 @@ def make_DFW_mb_model(
...
+    a = (T_rear - T_surface) / thickness
+    b = T_surface
+    return a * x + b
+
+
+def calculate_temperature_B(heat_flux, coolant_temp):
+    T_rear_tungsten = (
+        2.2e-5 * heat_flux + coolant_temp
+    )  # boron layers based off of rear temp of W mbs
+    return 5e-4 * heat_flux + T_rear_tungsten
```
Comment (RemDelaporteMathurin) [inline src/hisp/festim_models/mb_model.py:770]: @kaelyndunnell I plan to add documentation to these functions. Can you give me the references for these models?

Comment (RemDelaporteMathurin) [reply src/hisp/festim_models/mb_model.py:770]: Also, what is the justification of the boron temperature?  why 5e-4 * heat_flux + rear temperature?

Comment (kaelyndunnell) [reply src/hisp/festim_models/mb_model.py]: used this paper for tungsten temp model: 10.1088/1741-4326/ac2abc

File: src/hisp/festim_models/mb_model.py [hunk 2]
```diff
@@ -749,3 +753,156 @@ def make_DFW_mb_model(
...
+        heat_flux: heat flux in W/m2
+        coolant_temp: coolant temperature in K
+
+    Returns:
+        temperature in K
+    """
+    T_rear_tungsten = (
+        2.2e-5 * heat_flux + coolant_temp
+    )  # boron layers based off of rear temp of W mbs
+    return 5e-4 * heat_flux + T_rear_tungsten
```
Comment (kaelyndunnell) [inline src/hisp/festim_models/mb_model.py:796]: boron temp model, including 5e-4 coeff from thermal contact resistivity, here: https://doi.org/10.1088/1741-4326/acd9d9
using equation (4) and the value R_c produced from JET experiments

**Review by kaelyndunnell — CHANGES_REQUESTED** [review]

File: src/hisp/festim_models/mb_model.py [hunk 3]
```diff
@@ -749,3 +753,156 @@ def make_DFW_mb_model(
...
+    Args:
+        x: position in m
+        heat_flux: heat flux in W/m2
+        coolant_temp: coolant temperature in K
+        thickness: thickness of the W layer in m
+
+    Returns:
+        temperature in K
+    """
+    T_surface = 1.1e-4 * heat_flux + coolant_temp
```
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply src/hisp/festim_models/mb_model.py:774]: This exact equation is taken from Eq 11 from https://www.nature.com/articles/s41598-020-74844-w

File: src/hisp/festim_models/mb_model.py [hunk 4]
```diff
@@ -749,3 +753,156 @@ def make_DFW_mb_model(
...
+    T_rear = 2.2e-5 * heat_flux + coolant_temp
```
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply src/hisp/festim_models/mb_model.py:775]: If that so then I would make that calculation in python... Plus I don't see how @ehodille calculated the $T_\mathrm{rear}$ column here. Is it obtained from solving a heat transfer problem?

File: src/hisp/festim_models/mb_model.py (hunk 2 above)
Comment (RemDelaporteMathurin) [reply src/hisp/festim_models/mb_model.py:796]: Can you provide the full calculation? Eq 4 just gives the expression of the global thermal resistnace of the Be layer.

It should be a simple enough calculation, we need to calculate `2.2e-5` and `5e-4` programmatically, and only hardcode numreical values we can find in the paper.

image

Comment (kaelyndunnell) [reply src/hisp/festim_models/mb_model.py:796]: Yes! Equation is the same model used in equation (5), where the 150, 250, 400 values are the tungsten temperatures (detailed on page 15 & in table 2

Comment (kaelyndunnell) [reply src/hisp/festim_models/mb_model.py:796]: 5e-4 is a numerical value straight from the paper on page 14

Comment (kaelyndunnell) [reply src/hisp/festim_models/mb_model.py:796]: Noticed error in my code--should be taking T_surf instead of T_rear. will fix

Comment (RemDelaporteMathurin) [reply src/hisp/festim_models/mb_model.py]: Could you write a test to catch this?

File: src/hisp/festim_models/mb_model.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ehodille) [reply src/hisp/festim_models/mb_model.py:775]: > If that so then I would make that calculation in python... Plus I don't see how @ehodille calculated the T rear column here. Is it obtained from solving a heat transfer problem?

yes, with FESTIM and then looking at the temperature profile with paraview, if I remember correctly.

[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply src/hisp/festim_models/mb_model.py:775]: In 2D? Ok it's a bit hacky but I guess @kaelyndunnell 's approach here works as long as it's transparent.

**Pull request merged**